netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -667,6 +668,8 @@ net_ebpf_ext_layer_2_classify(
         net_xdp_ctx.base.data_end = packet_buffer + net_buffer->DataLength;
     }
 
+    old_irql = KeRaiseIrqlToDpcLevel();
```

Comment:
when would the old irql not be DPC here?

Reply:
If the NIC (or any underlying module) is running at an IRQL below dispatch level, then the WFP callouts inherits that IRQL. We're making gradual progress in shifting NICs, and networking in general, to run at passive (i.e., scheduler-friendly) IRQL.